core/translate/planner.rs

```diff
@@ -494,6 +484,49 @@ pub fn prepare_select_plan<'a>(schema: &Schema, select: ast::Select) -> Result<P
     }
 }
 
+pub fn prepare_delete_plan(
+    schema: &Schema,
+    tbl_name: &QualifiedName,
+    where_clause: Option<Expr>,
+) -> Result<Plan> {
+    let table_name = tbl_name.name.0.clone();
+
+    let table = if let Some(table) = schema.get_table(&table_name) {
+        table
+    } else {
+        crate::bail_parse_error!("Table {} not found", table_name);
+    };
+
+    let table_ref = BTreeTableReference {
+        table: table.clone(),
+        table_identifier: table_name.clone(),
+        table_index: 0,
+    };
+
+    // Parse and resolve the where_clause
+    let resolved_where_clauses = parse_where(where_clause, &[table_ref.clone()])?;
+
+    let plan = Plan {
+        source: SourceOperator::Scan {
+            id: 0,
+            table_reference: table_ref.clone(),
+            predicates: resolved_where_clauses.clone(),
+            iter_dir: None,
+        },
+        result_columns: vec![],
+        where_clause: resolved_where_clauses,
+        group_by: None,
+        order_by: None,
+        aggregates: vec![],
+        limit: None, // TODO: add support for limit
```

Comment:
This should be fairly simple to add, search for `DecrJumpZero` in `emitter.rs` and put one of those in `emit_delete_insns()`

Reply:
I want to maintain this PR small for now. And I want to improve my understanding on limit(+ order by). 

What do you think of I add a follow up PR right after?